packages/react-devtools-shared/src/__tests__/HooksTree-test.js

```diff
@@ -0,0 +1,92 @@
+describe('HooksTree', () => {
+  let React;
+  let ReactDOM;
+  let bridge: FrontendBridge;
+  let store: Store;
+  let utils;
+  let HooksTree;
+
+  let BridgeContext;
+  let InspectedElementContextController;
+  let StoreContext;
+  let TreeContextController;
+
+  beforeEach(() => {
+    utils = require('./utils');
+    utils.beforeEachProfiling();
+
+    bridge = global.bridge;
+    store = global.store;
+    store.collapseNodesByDefault = false;
+
+    React = require('react');
+    ReactDOM = require('react-dom');
+
+    BridgeContext = require('react-devtools-shared/src/devtools/views/context')
+      .BridgeContext;
+    InspectedElementContextController = require('react-devtools-shared/src/devtools/views/Components/InspectedElementContext')
+      .InspectedElementContextController;
+    StoreContext = require('react-devtools-shared/src/devtools/views/context')
+      .StoreContext;
+    TreeContextController = require('react-devtools-shared/src/devtools/views/Components/TreeContext')
+      .TreeContextController;
+
+    HooksTree = require('react-devtools-shared/src/devtools/views/Components/HooksTree')
+      .HooksTreeView;
+  });
+
+  const Contexts = ({
+    children,
+    defaultSelectedElementID = null,
+    defaultSelectedElementIndex = null,
+  }) => (
+    <BridgeContext.Provider value={bridge}>
+      <StoreContext.Provider value={store}>
+        <TreeContextController
+          defaultSelectedElementID={defaultSelectedElementID}
+          defaultSelectedElementIndex={defaultSelectedElementIndex}>
+          <InspectedElementContextController>
+            {children}
+          </InspectedElementContextController>
+        </TreeContextController>
+      </StoreContext.Provider>
+    </BridgeContext.Provider>
+  );
+
+  it('shows the complex value of custom hooks with sub-hooks', () => {
+    // props from `InspectedElementContext display complex values of useDebugValue: DisplayedComplexValue 1` snapshot
+
+    const container = document.createElement('div');
+    ReactDOM.render(
+      <Contexts>
+        <HooksTree
+          hooks={[
+            {
+              id: null,
+              isStateEditable: false,
+              name: 'DebuggableHook',
+              value: {
+                foo: 2,
+              },
+              subHooks: [
+                {
+                  id: 0,
+                  isStateEditable: true,
+                  name: 'State',
+                  value: 1,
+                  subHooks: [],
+                },
+              ],
+            },
+          ]}
+        />
+      </Contexts>,
+      container,
+    );
+
+    const hook = container.querySelector('.NameValueRow');
+    // it's actually DebuggableHook: {foo:2} but the first colon is added by css
+    // which isn't available in the test
+    expect(hook.textContent).toEqual('DebuggableHook{foo: 2}');
```

Comment:
This test seems kind of brittle.

I think we can probably just skip `HooksTree-test.js` :smile:

Reply:
That's the only test that is failing though Brian. HooksTree contains the bug. Why do you think the actual textual content is brittle? If you have an issue with the selector then I could add a `data-testid`. Since the relevant markup isn't semantic at all I don't see other opportunities

Reply:
Hard-coding a JSON object in a test is brittle. That object should be created by the backend. e.g. What happens if the backend structure changes in a way that would break things, but your test is written with the hard-coded JSON object?

I don't have time to dig deep on this right now unfortunately, so my feedback so far has been pretty shallow. In a few days I might have more time, but I need to wrap up some RFCs first.

If you disagree with my PR feedback so far, then let's just set it down for a while until I have time to really dig in.

Reply:
Ok I think I got the hang of how I can verify my devtools changes. This was a bit frustrating when starting out so apologies if this sounded more brisk than intended.

Reply:
No worries.